### DISCUSSION TITLE
Further help with opening encrypted files on a computer

### DISCUSSION THREAD

**Discussion by just-a-user-sudo** [gsantner/markor#1424] [Q&A] [answered]
Hello!

I enjoy Markor and appreciate that it has the ability to encrypt notes, but I'm only willing to encrypt them if I can decrypt them outside of the app. So, I tried doing that, starting with reading [this discussion](https://github.com/gsantner/markor/discussions/1372) and reading the READMEs [here](https://github.com/gsantner/markor) and [here](https://gitlab.com/opensource21/jpencconverter).

I didn't get much farther. Admittedly, I'm not familiar with Java— this is my first time trying to use Java software like this. 

First, I downloaded the source code for jpencconverter and extracted it to a folder. After that, I downloaded the latest version of Java SE on my Windows 10 computer from [here](https://www.oracle.com/java/technologies/javase-downloads.html) and installed it. The installer had already setup its own path variable, and I set a system variable to where Java is installed and named it "JAVA_HOME". And... that's all I could wrap my head around.

My question is, with Java installed and jpencconverter extracted on my computer, how exactly do I go about taking an encrypted *.md.jenc note from Markor and then decrypting it with jpencconverter? (I tried to answer this question with the READMEs and online searches but failed to.)

**Comment by gsantner** [comment]
@opensource21 is the best person to ask :-)

**Reply by opensource21** [reply]
I improved the Readme.md hopefully it's know clear enough.

**Comment by just-a-user-sudo** [comment, marked as answer]
Thank you @opensource21 for updating the  [README](https://gitlab.com/opensource21/jpencconverter). To answer my own question:

Don't download the source code, download a release by hovering over "Deployments" and clicking on "Releases" as shown: ![jenc png](https://user-images.githubusercontent.com/80051348/132092325-05a993d9-e59a-424e-9c0e-0ea2040708bb.png)

On the mobile site, bring out the menu, tap "Deployments", bring out the menu again and tap "Releases" as shown:

https://user-images.githubusercontent.com/80051348/132092606-48f355a1-6256-4031-8947-e0cc3e01a55e.mp4

Make sure you have Java installed on your computer. You can download the latest version from [here.](https://www.oracle.com/java/technologies/javase-downloads.html)

The README explains how to use jpencconverter.

After renaming `application.properties.sample` to `application.properties`, open it up as a text file. Edit the text after `filesearch.endir=` and `filesearch.dedir=` to point toward a directory within where you extracted the ZIP. (They can both point toward the same directory). For example, I have `filesearch.endir=notes` set, which means that notes in `C:\Users\*\UserApps\jpencconverter\notes` on my computer are eligible to be encrypted.

Removing `#`from `#password=*******` allows you to configure and save the default password input, otherwise you can leave it and insert it at every start.

As the README mentions, go into the folder you extracted the ZIP to with a command-shell and start with `java -jar <jar-file> <mode>`

Substitute `<jar-file>` with the filename of the .jar file in the extracted folder. For example, the current version is named `jpencconverter-0.2.1.jar`

Substitute `<mode>` with one of four modes as described in the README.
